### PR TITLE
chore: adding assertion of channel capabilities migration

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/27-interchain-accounts/controller/keeper/migrations.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+// NewMigrator returns a new Migrator.
+func NewMigrator(keeper *Keeper) Migrator {
+	return Migrator{keeper: keeper}
+}
+
+// AssertChannelCapabilityMigrations checks that all channel capabilities generated using the interchain accounts controller port prefix
+// are owned by the controller submodule and ibc.
+func (m Migrator) AssertChannelCapabilityMigrations(ctx sdk.Context) error {
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/keeper/migrations.go:24]: I'm open to suggestions on improving the name of this method.

File: modules/apps/27-interchain-accounts/controller/keeper/migrations.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	for _, channel := range m.keeper.GetAllActiveChannels(ctx) {
+		name := host.ChannelCapabilityPath(channel.PortId, channel.ChannelId)
+		owners, found := m.keeper.scopedKeeper.GetOwners(ctx, name)
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/controller/keeper/migrations.go]: I've updated to use `GetCapability` and added a call to `AuthenticateCapability` as well, but I actually think the authenticate call is redundant based on how `GetCapability` works underneath the hood. 

Forcing `isAuthenticated` to be false in the tests seems unreachable to me.

**Comment by damiannolan** [comment]
I believe we also need to call `SetMiddlewareEnabled()` for all active channels we iterate over in this migration handler. 
Any existing channels will need this set to continue to operate with the underlying app callbacks.

File: modules/apps/27-interchain-accounts/controller/keeper/migrations.go (hunk 1 above)
Comment (colin-axner) [reply modules/apps/27-interchain-accounts/controller/keeper/migrations.go:24]: I like the name

File: modules/apps/27-interchain-accounts/controller/keeper/migrations_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			err := SetupICAPath(path, ibctesting.TestAccAddress)
+			suite.Require().NoError(err)
+
+			tc.malleate()
+
+			migrator := keeper.NewMigrator(&suite.chainA.GetSimApp().ICAControllerKeeper)
+			err = migrator.AssertChannelCapabilityMigrations(suite.chainA.GetContext())
+
+			if tc.expPass {
+				suite.Require().NoError(err)
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/migrations_test.go:49]: want to add a check that the middleware enabled bool is set?

**Review by colin-axner — APPROVED** [review]
Excellent work. Super clean! Could you add a changelog entry?

File: modules/apps/27-interchain-accounts/controller/keeper/migrations.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/apps/27-interchain-accounts/controller/keeper/migrations.go]: I think it is okay to leave just to be extra verbose

**Comment by colin-axner** [comment]
> I believe we also need to call `SetMiddlewareEnabled()` for all active channels we iterate over in this migration handler. Any existing channels will need this set to continue to operate with the underlying app callbacks.

Makes sense to me. And if the chain removes the underlying app entirely, those middleware enabled checks won't be reached since the underlyingApp != nil check comes first

File: modules/apps/27-interchain-accounts/controller/keeper/migrations_test.go (hunk 3 above)
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/controller/keeper/migrations_test.go:49]: done 👍

**Review by AdityaSripal — APPROVED** [review]

**Pull request merged**